modules/services/fb_ai_features.js

```diff
@@ -173,56 +232,66 @@ export default {
             window.cancelIdleCallback(handle);
             _deferredAiFeaturesParsing.delete(handle);
         });
-        if (_tileCache && _tileCache.inflight) {
-            _forEach(_tileCache.inflight, abortRequest);
-        }
-        if (key !== undefined && _checkpoints.hasOwnProperty(key)) {
-            _graph = _checkpoints[key].graph;
-        }
-        else {
-            _graph = coreGraph();
-            _tree = coreTree(_graph);
-            _tileCache = { inflight: {}, loaded: {}, seen: {}, origIdTile: {} };
-        }
+
+        Object.values(_datasets).forEach(function(ds) {
+            if (ds.cache.inflight) {
+                Object.values(ds.cache.inflight).forEach(abortRequest);
+            }
+            ds.graph = coreGraph();
+            ds.tree = coreTree(ds.graph);
+            ds.cache = { inflight: {}, loaded: {}, seen: {}, origIdTile: {} };
+        });
+
+//todo: fix
```

Comment:
This TODO as well. anything need to be done here?

Reply:
As far as I can tell, the checkpoint stuff was there for the RapiD chapter of the walkthrough, but we don't need it now.  I setup the walkthrough graph as a separate dataset.  Now that we routinely juggle multiple datasets, adding another one for the walkthrough data is a piece of cake.